### PR TITLE
Keep debug info for debug and release with debug info build

### PR DESCRIPTION
**Pull request by Serge45**
## Brief ##
This PR adds ability to keep debug information for assembly kernels in debug and release with debug build.

## Details ##
 - Using `-g` or `-k` in `install.sh` will build whole library with assmembly debug information
 - For `TensileCreateLibrary`, use `--asm-debug` to enable it
 - For `Tensile`, simply add `AsmDebug: True` to tuning YAML's global parameter section.

## Misc ##
To check the debug information is kept or not, use `file <code_object_path>`, it shows

```
./release-debug/Tensile/library/TensileLibrary_gfx942.co: ELF 64-bit LSB shared object, *unknown arch 0xe0* version 1, dynamically linked, with debug_info, not stripped, too many notes (256)
```

for building with debug info. And it shows

```
./TensileLibrary_gfx942.co: ELF 64-bit LSB shared object, *unknown arch 0xe0* version 1, dynamically linked, not stripped, too many notes (256)
```
without debug info,

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -707,6 +710,10 @@ pushd .
     tensile_opt="${tensile_opt} -DTensile_LIBRARY_FORMAT=yaml"
   fi
 
+  if [[ "${asm_debug}" == true ]]; then
```
Comment (jichangjichang) [inline install.sh:713]: That checking build_release == false should be enough.
I didn't see the necessary to add asm_debug here unless you want to add another argument for install.sh to control the TENSILE_ASM_DEBUG directly.

**Comment by KKyang** [comment]
Isn't d stands for dependency?

File: install.sh (hunk 1 above)
Comment (Serge45) [reply install.sh:713]: Yea, you're right, will change soon.

**Comment by Serge45** [comment]
Yea, typo, I meant `-g`.

**Review by KKyang — APPROVED** [review]

**Review by jichangjichang — APPROVED** [review]
lgtm

**Pull request merged**